src/MarginTrading.NotificationGenerator.Services/TradingReportService.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using AutoMapper.Mappers;
+using Common.Log;
+using Lykke.Service.ClientAccount.Client;
+using MarginTrading.Backend.Contracts.Account;
+using MarginTrading.Backend.Contracts.AccountHistory;
+using MarginTrading.Backend.Contracts.TradeMonitoring;
+using MarginTrading.NotificationGenerator.Core;
+using MarginTrading.NotificationGenerator.Core.Domain;
+using MarginTrading.NotificationGenerator.Core.Services;
+using MarginTrading.NotificationGenerator.Core.Settings;
+using Microsoft.Extensions.Internal;
+
+namespace MarginTrading.NotificationGenerator.Services
+{
+    public class TradingReportService : ITradingReportService
+    {
+        private readonly IEmailService _emailService;
+        private readonly IConvertService _convertService;
+        private readonly ISystemClock _systemClock;
+        private readonly ILog _log;
+        private readonly TradingReportFilter _tradingReportFilter;
+
+        private readonly IClientAccountClient _clientAccountClient;
+        
+        private readonly Backend.Contracts.IAccountsApi _accountsApi;
+        private readonly Backend.Contracts.IAccountHistoryApi _accountHistoryApi;
+        private readonly Backend.Contracts.ITradeMonitoringReadingApi _tradeMonitoringReadingApi;
+
+        public TradingReportService(
+            IEmailService emailService,
+            IConvertService convertService,
+            ISystemClock systemClock, 
+            ILog log,
+            TradingReportFilter tradingReportFilter,
+            
+            IClientAccountClient clientAccountClient,
+            
+            Backend.Contracts.IAccountsApi accountsApi,
+            Backend.Contracts.IAccountHistoryApi accountHistoryApi,
+            Backend.Contracts.ITradeMonitoringReadingApi tradeMonitoringReadingApi)
+        {
+            _emailService = emailService;
+            _convertService = convertService;
+            _systemClock = systemClock;
+            _log = log;
+            _tradingReportFilter = tradingReportFilter;
+
+            _clientAccountClient = clientAccountClient;
+            
+            _accountsApi = accountsApi;
+            _accountHistoryApi = accountHistoryApi;
+            _tradeMonitoringReadingApi = tradeMonitoringReadingApi;
+        }
+
+        private bool LegalEntityPredicate(string legalEntity)
+        {
+            return string.IsNullOrWhiteSpace(_tradingReportFilter.LegalEntityRegex)
+                   || (!string.IsNullOrWhiteSpace(legalEntity)
+                       && Regex.IsMatch(legalEntity, _tradingReportFilter.LegalEntityRegex, RegexOptions.IgnoreCase));
+        }
+
+        public async Task PerformReporting()
+        {
+            var from = _systemClock.UtcNow.Date.AddMonths(-1);
+            var to = _systemClock.UtcNow.Date;
```

Comment:
I thought is should be report for previous calendar month, not report for last 30 days.

Reply:
Not an issue